File: NYTViewer-project/app/src/main/java/com/gmail/dp/denzay/nytviewer/NYTViewerApp.java
package com.gmail.dp.denzay.nytviewer;

import android.app.Application;

import com.gmail.dp.denzay.nytviewer.di.AppComponent;
import com.gmail.dp.denzay.nytviewer.di.AppModule;
import com.gmail.dp.denzay.nytviewer.di.DaggerAppComponent;

public class NYTViewerApp extends Application {

    private static AppComponent mAppComponent;

    @Override
    public void onCreate() {
        super.onCreate();
        mAppComponent = DaggerAppComponent.builder().appModule(new AppModule(this)).build();
    }


    public static AppComponent getAppComponent() {
        return mAppComponent;
    }

}

File: NYTViewerMVVM/app/src/main/java/com/gmail/dp/denzay/nytviewer_mvvm/NYTViewerApp.kt
package com.gmail.dp.denzay.nytviewer_mvvm

import android.app.Activity
import android.app.Application
import com.gmail.dp.denzay.nytviewer_mvvm.di.components.DaggerAppComponent
import dagger.android.AndroidInjector
import dagger.android.DispatchingAndroidInjector
import dagger.android.HasActivityInjector
import javax.inject.Inject

class NYTViewerApp: Application(), HasActivityInjector {

    @Inject
    lateinit var dispatchingAndroidInjector: DispatchingAndroidInjector<Activity>

    override fun activityInjector(): AndroidInjector<Activity> = dispatchingAndroidInjector

    override fun onCreate() {
        super.onCreate()

        DaggerAppComponent
            .builder()
            .setContext(this)
            .build()
            .inject(this)
    }
}